## Parkinson’s Disease Detection

### Parkinson’s disease is a progressive disorder of the central nervous system affecting movement and inducing tremors and stiffness. It has 5 stages to it and affects more than 1 million individuals every year.

### Unfortunately, this is chronic and has no cure yet. It is a neurodegenerative disorder affecting dopamine-producing neurons in the brain.

**XGBoost is a new Machine Learning algorithm designed with speed and performance in mind. XGBoost stands for eXtreme Gradient Boosting and is based on decision trees. In this project, we will import the XGBClassifier from the xgboost library; this is an implementation of the scikit-learn API for XGBoost classification.**

In [1]:
pip install numpy pandas sklearn xgboost

### 1. Recall the required libraries:

In [2]:
import numpy as np
import pandas as pd
import os, sys
from sklearn.preprocessing import MinMaxScaler
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

### 2. Upload the dataset & print the first 5 observations [rows] to get deeper understanding for the dataset. What the variables are about?

In [14]:
df = pd.read_csv('https://raw.githubusercontent.com/RewanEmam/Parkinson-s-Disease-in-Python/main/parkinsons.data', sep = ',', index_col=0)
df.head()

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...        D2       PPE
name                                       ...                    
phon_R01_S01_1      119.992       157.302  ...  2.301442  0.284654
phon_R01_S01_2      122.400       148.650  ...  2.486855  0.368674
phon_R01_S01_3      116.682       131.111  ...  2.342259  0.332634
phon_R01_S01_4      116.676       137.871  ...  2.405554  0.368975
phon_R01_S01_5      116.014       141.781  ...  2.332180  0.410335

[5 rows x 23 columns]

### 3. This function gives me more details about the type of each variable, number of columns & values, etc.

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 195 entries, phon_R01_S01_1 to phon_R01_S50_6
Data columns (total 23 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   MDVP:Fo(Hz)       195 non-null    float64
 1   MDVP:Fhi(Hz)      195 non-null    float64
 2   MDVP:Flo(Hz)      195 non-null    float64
 3   MDVP:Jitter(%)    195 non-null    float64
 4   MDVP:Jitter(Abs)  195 non-null    float64
 5   MDVP:RAP          195 non-null    float64
 6   MDVP:PPQ          195 non-null    float64
 7   Jitter:DDP        195 non-null    float64
 8   MDVP:Shimmer      195 non-null    float64
 9   MDVP:Shimmer(dB)  195 non-null    float64
 10  Shimmer:APQ3      195 non-null    float64
 11  Shimmer:APQ5      195 non-null    float64
 12  MDVP:APQ          195 non-null    float64
 13  Shimmer:DDA       195 non-null    float64
 14  NHR               195 non-null    float64
 15  HNR               195 non-null    float64
 16  status            195 non

### 4. Specify the features and labels values:

In [12]:
# Get the features and labels
features=df.loc[:,df.columns!='status'].values[:,1:]
labels=df.loc[:,'status'].values

### 5. Since the status variable [column] contains 2 types of values which are 0 & 1. I will count the number of zeros and ones:

In [15]:
# the count of each label (0 and 1) in labels
print(labels[labels==1].shape[0], labels[labels==0].shape[0])

147 48


**We have 147 ones and 48 zeros in the status column in our dataset.**

### 6. Initialize a MinMaxScaler and scale the features to between -1 and 1 to normalize them. The MinMaxScaler transforms features by scaling them to a given range. The fit_transform() method fits to the data and then transforms it.

In [16]:
# Scale the features to between -1 and 1
scaler=MinMaxScaler((-1,1))
x=scaler.fit_transform(features)
y=labels

### 7. Now, split the dataset into training and testing sets keeping 20% of the data for testing.

In [17]:
# Split the dataset
x_train,x_test,y_train,y_test=train_test_split(x, y, test_size=0.2, random_state=7)

### 8. Initialize an XGBClassifier and train the model. This classifies using eXtreme Gradient Boosting- using gradient boosting algorithms for modern data science problems. It falls under the category of Ensemble Learning in ML, where train and predict using many models to produce one superior output.

In [18]:
# Train the model
model=XGBClassifier()
model.fit(x_train,y_train)

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

### 9.Finally, generate y_pred (predicted values for x_test) and calculate the accuracy for the model. Print it out.

In [19]:
# Calculate the accuracy
y_pred=model.predict(x_test)
print(accuracy_score(y_test, y_pred)*100)

97.43589743589743


## I used an XGBClassifier for this and made use of the sklearn library to prepare the dataset. This gives us an accuracy of 94.44%, which is great considering the number of lines of code in this python project.